avocado/core/settings.py

```diff
@@ -107,6 +107,8 @@ def convert_value_type(value, value_type):
             value_type = float
         elif value_type == 'list':
             value_type = list
+        elif value_type == 'path':
+            value_type = os.path.expanduser
```

Comment:
Yep, I noticed the `AttributeError` thing later, thought I posted a note here, but seems it wasn't posted.

Anyway, I think we should always document exceptions that are raised by the code in context, and not really try to document everything that can go wrong within a function/method.

In this case, the `AttributeError` is caused by an incorrect type, even different than the one specific by the `value` parameter docstring.  As an example, what if users supply a `value` parameter that has the `startswith` attribute, but raises `FooError`?